### PR TITLE
Replace `rc` with `ini`

### PR DESCRIPTION
**Pull request by Gudahtt**
We use the `rc` package to read the `.metamaskrc` configuration file, which is in "ini" format. This package has been replaced by the `ini` package.

The `rc` package was not actively maintained, and it has had recent security vulnerabilities. But most importantly, the config object returned by `rc` includes a bunch of extra information that made [build script validation](https://github.com/MetaMask/metamask-extension/issues/15003) difficult to implement. Specifically, it  made it challenging to ensure no extra environment variables were present.

The `ini` package on the other hand is simple, well maintained, and is simpler to use. This package doesn't add any extra properties to the object it returns, making validation easy.

## Manual Testing Steps

This should have no functional changes.

## Pre-Merge Checklist

- [x] PR template is filled out
- [x] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [ ] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

### CODE REVIEW

File: lavamoat/build-system/policy.json [hunk 1]
```diff
@@ -4250,1314 +4250,547 @@
...
-        "gulp-watch>anymatch>micromatch>kind-of": true
-      }
-    },
-    "gulp-watch>chokidar>readdirp>micromatch>extglob>expand-brackets>define-property>is-descriptor>is-data-descriptor": {
-      "packages": {
-        "gulp-watch>anymatch>micromatch>kind-of": true
-      }
-    },
-    "gulp-watch>chokidar>readdirp>micromatch>extglob>expand-brackets>extend-shallow": {
-      "packages": {
```
Comment (brad-decker) [inline lavamoat/build-system/policy.json:4775]: We really gotta figure this out... everyone's systems are producing different policies.

File: yarn.lock [hunk 2]
```diff
@@ -14393,6 +14393,11 @@ ini@^1.3.4, ini@^1.3.5, ini@~1.3.0:
   resolved "https://registry.yarnpkg.com/ini/-/ini-1.3.8.tgz#a29da425b48806f34767a4efce397269af28432c"
   integrity sha512-JV/yugV2uzW5iMRSiZAyDtQd+nxtUnjeLt0acNdw98kKLrvuRVyB80tsREOE7yvGVgalhZ6RNXCmEHkUKBKxew==
 
+ini@^3.0.0:
+  version "3.0.0"
+  resolved "https://registry.yarnpkg.com/ini/-/ini-3.0.0.tgz#2f6de95006923aa75feed8894f5686165adc08f1"
+  integrity sha512-TxYQaeNW/N8ymDvwAxPyRbhMBtnEwuvaTYpOQkFx1nSeusgezHniEc/l35Vo4iCq/mMiTJbpD7oYxN98hFlfmw==
+
```
Comment (brad-decker) [inline yarn.lock:14400]: er, shouldn't there be a removal of rc? Or does one of our deps depend on it

File: lavamoat/build-system/policy.json (hunk 1 above)
Comment (brad-decker) [reply lavamoat/build-system/policy.json:4775]: @Gudahtt could you try wiping node_modules and regenerating to see if the diff lessens?

Comment (Gudahtt) [reply lavamoat/build-system/policy.json:4775]: Yeah It's not good. I tried restoring this via the policy override, but adding `fsevents` didn't trigger it to fill out the `fsevents` dependencies. So it wasn't really viable, we'd have to copy every deleted line here into the override to get it to stick.

File: yarn.lock (hunk 2 above)
Comment (Gudahtt) [reply yarn.lock:14400]: It's used by a number of our other dependencies. This was surprising to me too given that it's a dev tool but :shrug:

File: lavamoat/build-system/policy.json (hunk 1 above)
Comment (Gudahtt) [reply lavamoat/build-system/policy.json:4775]: I tried wiping out `node_modules`, adding `gulp-watch>chokidar>fsevents` to the override (it's the source of most of these removals), then reinstalling and re-running `lavamoat:build:auto`. No dice.

Comment (brad-decker) [reply lavamoat/build-system/policy.json:4775]: I noticed in one of my new prs that these are all removed as well, it doesn't even seem to be deterministic on the same machine with node_modules fresh. Oh well, a problem for a different PR

**Review by brad-decker — DISMISSED** [review]
LGTM

**Review by darkwing — DISMISSED** [review]

**Review by brad-decker — APPROVED** [review]

**Comment by Gudahtt** [comment]
Rebased to resolve a conflict in the LavaMoat policy file. I have tried a clean install + `lavamoat:build:auto` and no further changes appeared, so this should be good to go again now.

**Review by darkwing — APPROVED** [review]

**Pull request merged**